tests/returntypes/test_new_return_types_qnode.py

```diff
@@ -3005,6 +3020,7 @@ def circuit(a):
 
         with tf.GradientTape() as tape:
             out = circuit(x)
+            out = tf.concat([tf.reshape(i, [-1]) for i in out], 0)
```

Comment:
We can use `tf.experimental.numpy.hstack()` here to keep things cleaner. Same for the other places below. 

There are also other places that are already merged, but maybe we can open a separate PR for that instead of doing it here.

Reply:
Let me replace that in this PR 👍